crates/mpz-garble/src/protocol/deap/mod.rs

```diff
@@ -139,6 +139,86 @@ impl DEAP {
         self.state.lock().unwrap()
     }
 
+    /// Commits the provided input values.
+    pub async fn commit<Ctx, OTS, OTR>(
+        &self,
+        ctx: &mut Ctx,
+        inputs: &[ValueRef],
+        ot_send: &mut OTS,
+        ot_recv: &mut OTR,
+    ) -> Result<(), DEAPError>
+    where
+        Ctx: Context,
+        OTS: OTSendEncoding<Ctx> + Send,
+        OTR: OTReceiveEncoding<Ctx> + Send,
+    {
+        let assigned = self.state().memory.drain_assigned(inputs);
+        match self.role {
+            Role::Leader => {
+                try_join!(
+                    ctx,
+                    self.gen
+                        .setup_assigned_values(ctx, &assigned, ot_send)
+                        .map_err(DEAPError::from),
+                    self.ev
+                        .setup_assigned_values(ctx, &assigned, ot_recv)
+                        .map_err(DEAPError::from)
+                )??;
+            }
+            Role::Follower => {
+                try_join!(
+                    ctx,
+                    self.ev
+                        .setup_assigned_values(ctx, &assigned, ot_recv)
+                        .map_err(DEAPError::from),
+                    self.gen
+                        .setup_assigned_values(ctx, &assigned, ot_send)
+                        .map_err(DEAPError::from)
+                )??;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Commits the provided values for proving.
+    pub async fn commit_prove<Ctx, OTR>(
+        &self,
+        ctx: &mut Ctx,
+        values: &[ValueRef],
+        ot_recv: &mut OTR,
+    ) -> Result<(), DEAPError>
+    where
+        Ctx: Context,
+        OTR: OTReceiveEncoding<Ctx> + Send,
+    {
+        let assigned = self.state().memory.drain_assigned(values);
+        self.ev
+            .setup_assigned_values(ctx, &assigned, ot_recv)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Commits the provided values for verifying.
+    pub async fn commit_verify<Ctx, OTS>(
+        &self,
+        ctx: &mut Ctx,
+        values: &[ValueRef],
+        ot_send: &mut OTS,
+    ) -> Result<(), DEAPError>
+    where
+        Ctx: Context,
+        OTS: OTSendEncoding<Ctx> + Send,
+    {
+        let assigned = self.state().memory.drain_assigned(values);
```

Comment:
Can we have a check that only blind values can be committed to for verification?

Reply:
Based on our DMs adding these checks requires other changes also. This part of the code will be obsoleted soon, so we decided to not make any changes at this time.